### PR TITLE
app/user: ensure that user gets sanitized while calling plugin hooks

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Makes sanitization more explicit, the only reason we got away from this so far was assigning the pointer to the new value. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42647

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by amyblais** [comment]
Which versions does this affect?

**Comment by isacikgoz** [comment]
@amyblais this specific change is for peace of mind but we are still investigating other potential leakages.

**Review by srkgupta — APPROVED** [review]
Thanks @isacikgoz. As discussed, although this is not absolutely necessary, but as you mentioned that it gives peace of mind. Tested the changes before & after the changes and nothing is changed/affected. And thanks for adding the test.

File: app/user_test.go [hunk 1]
```diff
@@ -200,24 +200,70 @@ func TestCreateUser(t *testing.T) {
...
+			Email:       model.NewId() + "success+test@example.com",
+			Nickname:    "Darth Vader",
+			Username:    "vader" + model.NewId(),
+			Password:    "passwd12345",
+			AuthService: "",
+		}
+		_, err := th.App.CreateUser(th.Context, user)
+		require.Nil(t, err)
+
+		time.Sleep(1 * time.Second)
```
Comment (agnivade) [inline app/user_test.go:261]: Any reason we are sleeping here?

File: app/user.go [hunk 2]
```diff
@@ -288,7 +288,7 @@ func (a *App) createUserOrGuest(c *request.Context, user *model.User, guest bool
 		a.Srv().Go(func() {
 			pluginContext := pluginContext(c)
 			pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
-				hooks.UserHasBeenCreated(pluginContext, user)
+				hooks.UserHasBeenCreated(pluginContext, ruser)
```
Comment (agnivade) [inline app/user.go:291]: It's interesting to see that this never had a race condition. We are accessing a pointer in a separate goroutine here. Ideally, we should clone the user before sending it to the function. Anyways, out of scope for this issue.

File: app/user_test.go (hunk 1 above)
Comment (isacikgoz) [reply app/user_test.go:261]: The hook is not getting called right away, as it runs on a separate thread. And other tests was structured the same way.

File: app/user.go (hunk 2 above)
Comment (isacikgoz) [reply app/user.go:291]: I think the reason we never had a race here because we don't write to that pointer after it passed here.

**Review by agnivade — APPROVED** [review]

**Pull request merged**